### PR TITLE
feat(roocode): add RooCode integration (configurator, slash commands, templates)

### PR DESCRIPTION
**Pull request by jax-max**
#feat(roocode): add RooCode integration (configurator, slash commands, templates)
## Description
This PR adds full integration of the RooCode tool into the OpenSpec ecosystem, enabling seamless usage of RooCode's capabilities through the existing toolchain.

## Changes Included
- Add RooCode configurator class (core logic for tool initialization and configuration)
- Register RooCode to the tool registry (enable discovery and loading of the tool)
- Implement RooCode template files (standardized templates for RooCode-related operations)
- Add RooCode slash command support (allow triggering RooCode via slash commands)
- Update README documentation (add usage guide and integration notes for RooCode)
- Add relevant test cases (cover core functionalities and edge scenarios of RooCode integration)

## Checklist
- [x] Code follows project coding conventions
- [x] All new features are covered by test cases
- [x] Documentation is updated to reflect changes
- [x] No breaking changes to existing functionalities
- [x] Dependencies (if any) are properly declared

## Notes
- RooCode integration maintains compatibility with existing toolchain workflows
- Slash commands follow the project's established command naming convention
- Test cases include unit tests for the configurator class and integration tests for tool registration/command execution

Feel free to let me know if any adjustments are needed for the integration logic or documentation!

## Summary by CodeRabbit

* **New Features**
  * RooCode added as a supported AI tool with native slash-command support.
  * RooCode generates OpenSpec slash commands for proposal, apply, and archive workflows in `.roo/commands/`.
  * Generated commands support safe refresh via `openspec update`.

* **Documentation**
  * Updated to include RooCode in supported AI tools table.

### CODE REVIEW

**Review by Israel-Laguan — APPROVED** [review]

File: openspec/specs/cli-init/spec.md [hunk 1]
```diff
@@ -91,6 +91,12 @@ This project uses OpenSpec to manage AI assistant workflows.
 <!-- OPENSPEC:END -->
 ```
 
+#### Scenario: Configuring RooCode
+
+- **WHEN** RooCode is selected
+- **THEN** create or update `ROOCODE.md` in the project root directory (not inside `openspec/`)
```
Comment (TabishB) [inline openspec/specs/cli-init/spec.md]: @jax-max I believe roocode supports the `AGENTS.md` can we get away with not having to create another rules file for roo if not needed?

Comment (ravshansbox) [reply openspec/specs/cli-init/spec.md]: @TabishB AFAIK claude also supports AGENTS.md, maybe we can get rid of CLAUDE.md as well?

Comment (jax-max) [reply openspec/specs/cli-init/spec.md]: Thanks for the heads-up. Roocode doesn’t have a native mechanism to "default scan ROOCODE.md and auto-apply its rules". I’ve removed the ROOCODE.md-related mechanism and only retained the slash command functionality – please help review, much appreciated!
 @TabishB

Comment (TabishB) [reply openspec/specs/cli-init/spec.md]: @ravshansbox There's currently only 2 coding agent that don't support `AGENTS.md`. Claude and Cline. Cline will support it soon, but I don't think Claude will haha. it's been requested for a while: https://github.com/anthropics/claude-code/issues/6235
